src/core/src/op/depth_to_space.cpp

```diff
@@ -42,8 +42,8 @@ std::shared_ptr<Node> DepthToSpace::clone_with_new_inputs(const OutputVector& ne
 void DepthToSpace::validate_and_infer_types() {
     OV_OP_SCOPE(v0_DepthToSpace_validate_and_infer_types);
 
-    const auto output_shape = shape_infer(this, ov::util::get_node_input_partial_shapes(*this)).front();
-    set_output_type(0, get_input_element_type(0), output_shape);
+    const auto output_shapes = shape_infer(this, ov::util::get_node_input_partial_shapes(*this));
```

Comment:
```suggestion
    const auto& output_shapes = shape_infer(this, ov::util::get_node_input_partial_shapes(*this));
```

Reply:
All usage aligned to use `const auto` vector is created in this place the `&` is not required.